examples/bearriver/src/FRP/BearRiver.hs

```diff
@@ -175,17 +175,20 @@ afterEachCat = afterEachCat' 0
 
 
 -- * Events
-
 instance Functor Event where
-  fmap f NoEvent   = NoEvent
-  fmap f (Event c) = Event (f c)
+   f `fmap` (Event c) = Event (f c)
+   _ `fmap` _         = NoEvent
 
 instance Applicative Event where
   pure = Event
-
   Event f <*> Event x = Event (f x)
   _       <*> _       = NoEvent
 
+instance Monad Event where
```

Comment:
If you want you can add a haddock in the line before that says that this instance is the same as `Maybe`'s.